Library.DAL/Repositories/CountryRepository.cs

```diff
@@ -1,12 +1,21 @@
 ﻿using Library.DAL.Entities;
+using Library.DAL.Repositories.Interfaces;
 using Library.DAL.UnitOfWork;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Library.DAL.Repositories
 {
-    public class CountryRepository : EfCoreRepository<Country, ApplicationContext>
+    public class CountryRepository : EfCoreRepository<Country, ApplicationContext>, ICountryRepository
     {
         public CountryRepository(ApplicationContext context) : base(context)
         {
         }
+
+        public IQueryable<Country> GetCountriesByNameFragement(string countryNameFragment)
```

Comment:
In this case, you can use IEnumarable
IQueryble is used for the non-constant "where" statement

Reply:
IQueryable<T> Find(Expression<Func<T, bool>> predicate = null);
IQueryable<T> FindAll();

public IQueryable<T> Find(Expression<Func<T, bool>> predicate = null)
{
    return this.Set.Where(predicate); // Set is DBSet
}

and somewhere in the code .ToList() or .FirstOrDefault(), etc.